knimin/tests/test_ag_third_party.py

```diff
@@ -9,19 +9,21 @@
 from knimin.tests.tornado_test_base import TestHandlerBase
 from knimin import db
 from knimin.handlers.ag_third_party import ThirdPartyData, NewThirdParty
+from functools import partial
 
 
 class AGThirdPartyHandler(TestHandlerBase):
     ext_survey_fp = join(dirname(realpath(__file__)), 'data',
                          'external_survey_data.csv')
+    data_vioscreen = {'name': 'Vioscreen',
+                      'description': 'FFQ',
+                      'url': 'http://vioscreen.com'}
+    _clean_up_funcs = []
 
-    def setUp(self):
-        # Make sure vioscreen survey exists in DB
-        try:
-            db.add_external_survey('Vioscreen', 'FFQ', 'http://vioscreen.com')
-        except ValueError:
-            pass
-        super(AGThirdPartyHandler, self).setUp()
+    def tearDown(self):
+        for f in self._clean_up_funcs:
+            f()
```

Comment:
Awesome! I just realized that depending on where the test failed this call may also fail - probably safer adding a try/except around in which we just print on the screen that the clean up failed? This way if there is more than one call on the clean up all of them get executed.